From 4f9692a791037f144a1e17e074200e752ee73b64 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20=C4=8Cern=C3=BD?= <jcerny@redhat.com>
Date: Mon, 30 Jan 2023 11:38:23 +0100
Subject: [PATCH 08/22] Fix a leaked variable

Addressing:
openscap-1.3.7/utils/oscap-ds.c:403: leaked_storage: Variable "temp_cwd" going out of scope leaks the storage it points to.
---
 utils/oscap-ds.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/utils/oscap-ds.c b/utils/oscap-ds.c
index f5d8ca8b7..2ef18a8e7 100644
--- a/utils/oscap-ds.c
+++ b/utils/oscap-ds.c
@@ -368,6 +368,8 @@ int app_ds_sds_compose(const struct oscap_action *action) {
 	char* temp_cwd = strdup(action->ds_action->file);
 	char *temp_cwd_dirname = oscap_dirname(temp_cwd);
 	if (chdir(temp_cwd_dirname) < 0) {
+		free(temp_cwd_dirname);
+		free(temp_cwd);
 		goto cleanup;
 	}
 	free(temp_cwd_dirname);
-- 
2.34.1

